### ISSUE TITLE
Incorrect (unsound) computation in certification function

### ISSUE THREAD

**Issue by tobycmurray** [klasleino/gloro#9]
After calculating the model's output `y`, computing its argmax `j` (with corresponding logit value `y_j`) and margin Lipschitz constants `K_ij` the certification routine does the following (whose intention is explained in the comment below):
 https://github.com/klasleino/gloro/blob/a218dcdaaa41951411b0d520581e96e7401967d7/gloro/models/gloronet.py#L277-L286

Unfortunately, this replaces *every* element of `y_bot_i` where `y` is equal to `y_j` with `-inf`. This can result in unsound certification.

Consider a trivial case in which a binary classifier outputs logits that are equal to 0 (e.g. for an input point on the decision boundary). In this case, both logits will be replaced by `-inf` by the above code. Thus the value of the "bottom" logit will also be `-inf` and the output point will be incorrectly certified robust.

Instead the code above should be modified to replace only the logit `j` in `y_bot_i` with `-inf`.
